cloud-azure/src/test/java/com/sequenceiq/cloudbreak/cloud/azure/AzureEncryptionResourcesTest.java

```diff
@@ -296,6 +296,8 @@ public void testCreateDiskEncryptionSetShouldReturnExistingDiskEncryptionSetWith
         when(azureClient.getCurrentSubscription()).thenReturn(subscription);
         when(azureClient.getDiskEncryptionSetByName(any(String.class), any(String.class))).thenReturn(des);
         when(azureClient.keyVaultExists("dummyResourceGroup", "dummyVaultName")).thenReturn(Boolean.TRUE);
+        when(azureClient.checkKeyVaultAccessPolicyForServicePrincipal("dummyResourceGroup", "dummyVaultName", DES_PRINCIPAL_ID))
+                .thenReturn(true);
```

Comment:
Please add also a test case where `checkKeyVaultAccessPolicyForServicePrincipal()` returns `false`. See `testCreateDiskEncryptionSetShouldReturnNewlyCreatedDiskEncryptionSetIfNotAlreadyExistsAndGrantKeyVaultAccessPolicyError()` that simulates the failure of `azureClient.grantKeyVaultAccessPolicyToServicePrincipal()`. The retry logic itself does not have to be simulated, what matters is the outcome `ActionFailedException` that contains the right cause, which is verified there by `verifyActionFailedException()`.